src/test/suite/tokenizer.test.ts

```diff
@@ -1,8 +1,15 @@
 import * as assert from "assert";
 import { tokenize } from "../../core/tokenizer";
 import { flatten, range } from "lodash";
+import { SupportedLanguageId } from "../../languages/constants";
 
 type TestCase = [string, string[]];
+type LanguageTokenizerTests = Partial<
+  Record<
+    SupportedLanguageId,
+    { tests: TestCase[]; exclusionPredicate?: (input: string) => boolean }
+  >
+>;
```

Comment:
I think I'd have this type be the smaller one, and then add a docstring on `exclusionPredicate`, because it's not obvious what that does 😅
```suggestion
/** Language-specific tokenizer test configuration object */
interface LanguageTokenizerTests {
   /** Language-specific test cases to run in addition to the global tests for this language */
   additionalTests: TestCase[];

   /**
    * By default we run all global tests in the given language, in addition to the specific tests.  We exclude global test inputs that match this predicate.
    */
   exclusionPredicate?: (input: string) => boolean;
}
```